tests/compile-fail/while_loop.rs

```diff
@@ -0,0 +1,49 @@
+#![feature(plugin)]
+#![plugin(clippy)]
+
+#[deny(while_let_loop)]
+fn main() {
+    let y = Some(true);
+    loop { //~ERROR
+        if let Some(_x) = y {
+            let _v = 1;
+        } else {
+            break;
+        }
+    }
+    loop { //~ERROR
+        if let Some(_x) = y {
+            let _v = 1;
+        } else {
+            break
+        }
+    }
+    loop { // no error, break is not in else clause
```

Comment:
Issue with this test:
```
loop { // no error, match is not the only statement
    match y {
        Some(_x) => true,
        None => break
    };
    let _x = 1;
}
```
This **should** return a warning, because it _can_ be replaced with `while let`. The only time there are problems is when there are statements before the `match`.